### ISSUE TITLE
Font issues

### ISSUE THREAD

**Issue by guekho64** [JFormDesigner/FlatLaf#309]
### It seems FlatLaf has some issues trying to detect which are my current font settings, as demonstrated by this simple example:

<details><summary>This is how it looks</summary>

![nope](https://user-images.githubusercontent.com/8020999/115145850-5bd14500-a019-11eb-893a-42b66427e36b.png)

</details>

<details><summary>This is how it <i><b>should</b></i> look</summary>

![ok](https://user-images.githubusercontent.com/8020999/115145881-90450100-a019-11eb-8156-6abd9098d09c.png)

</details>

Here's the code of this example (I had to modify FlatLaf src to access LinuxFontPolicy#getGnomeFont)

``` java
public class Main {
	public static void main (final String[] argv) {
		FlatIntelliJLaf.install();
		Font info = LinuxFontPolicy.getGnomeFont();
		String text = "This is what FlatLaf thinks it's my system wide font:\n\n";
		JOptionPane.showOptionDialog(
				null,
				text + info.toString(),
				"Correct Font Example",
				0,
				JOptionPane.INFORMATION_MESSAGE,
				null,
				new Object[] {"A button, of course"}, 0
				);
	}
}

```

#### FlatLaf Versions Tested

FlatLaf (master)
FlatLaf (1.1.2)

#### Desktop Environment

MATE 1.24.0

#### java -version
openjdk version "14.0.2" 2020-07-14
OpenJDK Runtime Environment (build 14.0.2+12-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 14.0.2+12-Ubuntu-120.04, mixed mode, sharing)

#### lsb_release -a
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal

#### uname -a
Linux PC 5.8.0-50-lowlatency 56~20.04.1-Ubuntu SMP PREEMPT Tue Apr 13 01:48:06 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**Comment by DevCharly** [comment]
I wonder what font is used in other Lafs.

Could you replace:

~~~java
FlatIntelliJLaf.install();
Font info = LinuxFontPolicy.getGnomeFont();
~~~

with

~~~java
UIManager.setLookAndFeel( UIManager.getSystemLookAndFeelClassName() );
Font info = UIManager.getFont( "Label.font" );
~~~

and

~~~java
UIManager.setLookAndFeel( UIManager.getCrossPlatformLookAndFeelClassName() );
Font info = UIManager.getFont( "Label.font" );
~~~

and post the two screenshots? Thanks.

**Comment by kingthorin** [comment]
I don't mean to derail this issue so if this isn't closely enough related feel free to let me know and I'll delete my comments and open a new issue (or take the discussion elsewhere if there's a Google group or something you'd rather I used).

What's the "proper" way to override FlatLaf's default fonts? We currently do this: `                 UIManager.getLookAndFeelDefaults().put("defaultFont", font);` but it doesn't seem to work.

**Comment by DevCharly** [comment]
@kingthorin please open a new issue

**Comment by guekho64** [comment]
Here you go:

```

SystemLAF = javax.swing.plaf.metal.MetalLookAndFeel
javax.swing.plaf.FontUIResource[family=Dialog,name=Dialog,style=bold,size=12]

CrossplatformLAF = javax.swing.plaf.metal.MetalLookAndFeel
javax.swing.plaf.FontUIResource[family=Dialog,name=Dialog,style=bold,size=12]


```

And yeah, I was expecting that, since Metal doesn't follow my setting neither....

**Comment by DevCharly** [comment]
Many thanks.

Yes, was expecting that for Metal too, but I actually thought that GTK is used as System Laf on Linux.

Could you please also try:
~~~java
UIManager.setLookAndFeel( new GTKLookAndFeel() );
~~~

I also wonder whether the text in the JButton is totally out of vertical center for Metal and GTK Lafs as in your first screenshot

**Comment by guekho64** [comment]
Here's how JOptionPane looks in GTK laf:

![image](https://user-images.githubusercontent.com/8020999/115463038-f8017480-a1f0-11eb-950c-feb4889bd45b.png)

Surprisingly enough, it doesn't acquire correct fonts either.

Here you can compare how it should look (MATE's oficial file browser: Caja 1.24)

![image](https://user-images.githubusercontent.com/8020999/115464808-40ba2d00-a1f3-11eb-9f7f-f336df8e0979.png)

Honestly, I'm not sure where could be the fault:

* Is java incorrectly parsing the output of `defaultToolkit().getDesktopProperty()`?
* MATE Desktop Environment is doing something wrong when reporting its default font to Java?
* Are some fonts simply misconfigured (Another case would be `Nimbus Sans Bold`, which reports itself as `Nimbus Sans-Bold`)?
  * Not to mention `SF Compact Rounded Semibold` was designed by ®Apple
* Font name conventions aren't strict enough, to allow for a trustworthy font metadata parsing?
  * For instance, doing `new Font("SF Compact Rounded Semibold")` actually works.
  * Doing `new Font("SF Compact Rounded Semi-bold")` fails. 
  * My PR #310 does not follow any conventions at all, and only relies in U+0020 (" ") as the only word delimiter available, and considers last token from `defaultToolkit().getDesktopProperty()` as fontsize; aside from that, it doesn't make any more assumptions, and tries **real hard** to guess which font it's currently in use, but of course, it's not infallible. As a fallback, it'll default to generate a `Dialog` font, which it's universally available to Java, being a Logic family font.

**Comment by guekho64** [comment]
Are there any news on this, or #310, by any chance?

**Comment by DevCharly** [comment]
Had some issues with PR #310...

There is a fix in commit 9edaf5892949523096a1270e45b16b06e4fb4d68
Please give it a try...
